### DISCUSSION TITLE
2.0 is out!

### DISCUSSION THREAD

**Discussion by Gissio** [Gissio/radpro#94] [Announcements]
The long-awaited Rad Pro 2.0 is out, with support for several devices: FS2011, Bosean FS-600, FS-1000, FS-5000 and FNIRSI GC-01.

The UI has been updated to the [OpenBridge 5.0](https://www.openbridge.no/) design system, improving legibility under sunlight.

The default conversion factor for J613/J614 tubes has also been changed to 30 cpm/µSv/h, improving accuracy.

The FNIRSI GC-01 display panel settings (added in 2.0rc5) have been removed. And the granularity and range of the "SET tubeHVFrequency" and "SET tubeHVDutyCycle" data communication requests have been increased, to CPU clock resolution and 100 Hz-100 kHz and 0 %-100 %, respectively (not stored on power-off).

Enjoy!

**Comment by Montecri** [comment]
Great milestone!!

A heads-up: I just installed it on my FS-5000 and the way the script bosean-fs5000-install.bat/flashtool.bat is written is making it flash radpro-bosean-fs5000-landscape-2.0.bin instead of the non-landscape one. Had to force the correct firmware manually.

And, perhaps the landscape thing can be one of the Display options? So a single firmware would satisfy both.

Great work! Thank you!

- Cristiano

**Reply by Gissio** [reply]
I just checked the scripts and found no error. Are you sure?

**Reply by Montecri** [reply]
Ran this on cmd:

D:\Downloads\radpro-flashtool-2.0\bosean>bosean-fs5000-install.bat

At the end I had the screen rotated by 90 degrees.

Removed radpro-bosean-fs5000-landscape-2.0.bin from the firmware directory and ran the above .bat again (didn't type anything, just used arrow key to select last command). This time it flashed the non-rotated screen.

Thanks!

**Reply by Gissio** [reply]
That was very helpful in locating the error: it was a file naming issue. Fixed.

**Comment by jonwhite227** [comment]
Hi,
I am getting an error when trying to flash RadPro 2.0 on my GC-01 with wch processor, which is currently running 2.0rc5 - the same error occurs when flashing from 2.0rc4. I'm using same STLink setup as used in all previous testing, but the script halts as below (image)

![2024-06-10_01-55-18](https://github.com/Gissio/radpro/assets/165075342/1344146f-c259-4459-b0f7-027c46186bda)

Any thoughts would be very welcome.

Jon

**Reply by Gissio** [reply]
Please copy the .bat file contained in the following file to the `fnirsi-gc01` folder of the flashtool and start it from there. It will provide extended debugging info. Copy the resulting output, paste it to a text file and upload it here.

[fnirsi-gc01-ch32f103r8t6-install-debug.zip](https://github.com/user-attachments/files/15815050/fnirsi-gc01-ch32f103r8t6-install-debug.zip)

**Reply by Gissio** [reply]
Could you test [https://github.com/Gissio/radpro/releases/tag/2.0.1beta3](https://github.com/Gissio/radpro/releases/tag/2.0.1beta3)?

**Reply by jonwhite227** [reply]
Yes, no problem.

**Reply by jonwhite227** [reply]
2.0.1beta3 version installs flawlessly on my GC-01. All functions are working okay, and stable after a couple of hours testing. Thank you for this update, let me know if I can be of further help.

**Comment by jonwhite227** [comment]
Here is text file as requested.

Jon
[GC-01 debug text radpro 2.0 jonwhite227.txt](https://github.com/user-attachments/files/15825711/GC-01.debug.text.radpro.2.0.jonwhite227.txt)

**Reply by Gissio** [reply]
Sorry: could you download the .zip file from my previous post again, and try again?

**Reply by jonwhite227** [reply]
Here you go - this one is 28kb instead of 14kb

[GC-01 debug text radpro 2.0 jonwhite227 ver. 2.txt](https://github.com/user-attachments/files/15826402/GC-01.debug.text.radpro.2.0.jonwhite227.ver.2.txt)

**Comment by Gissio** [comment]
Thanks to your dump I found the error! Will be fixed in the upcoming release.

**Reply by jonwhite227** [reply]
Glad I could help! 🙂

**Reply by kd5byb** [reply]
Having the same issue here with my FNIRSI GC-01.  None of the easy firmware loading methods work, so I resorted to an STLink clone dongle from Amazon.

I'm getting the same "halted due to debug-request" error:
![image](https://github.com/user-attachments/assets/f57b20d7-4bdf-4545-a582-9049c059a35d)

I notice you asked JonWhite to run the debug uploading tool and post the results.  Here's the results I'm getting.

I also tried downloading the link you posted about...doesn't seem to exist anymore.  

Any / all help appreciate!

thanks much,
ben

[kd5byb GC-01 firmware debug dump.txt](https://github.com/user-attachments/files/16646535/kd5byb.GC-01.firmware.debug.dump.txt)

**Comment by gc01noob** [comment]
easiest way to load the firmware:
use Linux eg Ubuntu (boot from USB live if you don't have it installed)

1. connect the device
2. open the BOOTLOADER drive
3. copy over the correct firmware for your MCU
4. leave the window open
4. copy over the firmware a 2nd time and click Yes to overwrite

viola your device will restart with radpro

(don't ask me why but it just works 100% of the time, at least for the GC01 - easiest to use a Linux with GUi like Gnome, if you only have a terminal check the discussion for the specific command to, cp isn't enough, if i remember right you have to cat)

**Reply by kd5byb** [reply]
I tried this the other evening - using a live-usb of Lubuntu I booted up my PC and no combination of holding down a button and powering on would make a new drive appear.  :(  I'm wondering if my FNIRSI GC-01 may be slighty different from others.  :(

I do appreciate the reply!

thanks,
ben

**Reply by gc01noob** [reply]
no need to hold any button, just power it on and the drive should appear

not sure what kernel Lubuntu uses (or how it handles /media/ usb drives), normal Ubuntu 22.04 has the driver so I'd try that

**Reply by kd5byb** [reply]
So just downloaded and made me a live USB key with 22.04 LTS.  Same behavior - won't open any sort of drive.  :(  

Was worth a shot for sure!

thanks much,
ben

**Reply by gc01noob** [reply]
aww that sux!

maybe sounds silly but have you tried a different USB cable? some are dodgy with some devices data even if they charge ok

does the terminal command lsusb show the device at all?
https://linuxhandbook.com/list-usb-devices/

**Reply by kd5byb** [reply]
Yeah, have tried several cables.  INTERESTINGLY, the first one I grabbed was indeed BAD!  (and got chucked)  Been a minute since I've had a bad USB cable.  I did try lsusb earlier inside 22.04 LTS and there's nothing showing.  And unplugging it and re-running lsusb makes no difference.  Interestingly, Win10 tries to find it, but has a strange USB error like it tried to mount it but couldn't?  :(

I'm not real smart with Github - should the RC3 release that worked on John's GC-01 be somewhere here?  I tried looking, didn't find it...

thanks much,
ben

**Reply by gc01noob** [reply]
unfortunatley if it doesn't show in lsusb you won't be able to do anything over USB
If the PC's USB port is also ok I'd suspect a faulty device (just to repeat you musn't press any button when you switch it on, just power on and connect)

btw radpro also let's you connect with USB to use the graphing/data capture software so you'll want it to at least show as a USB device even if you have to use a serial flasher. the company's own support video shows firmware updates via a mounted USB drive...

are you within warranty to return it?

**Reply by kd5byb** [reply]
Well...it gets more odd.  I decided to try Linux again...and it will open a drive called "GC01 BOOT".  I can copy the firmware file there, and it seems to copy okay.  I can't copy it a second time...it says there's not enough disk space.  I saw the other post about using cat to copy it the second time (or both times) and that appears to work...but the unit doesn't restart and when I power cycle it, I do see the drive again with the "Readme.txt" (completely blank!).  VERY STRANGE.

**Reply by jonwhite227** [reply]
I had the same issues with the usb window, and then an unresponsive device. The easiest solution going forward is to solder a 4 pin header to the gc-01 board and then use an STLink adaptor to run the update. This also makes any future updates or testing a whole lot easier. Info should all be on here, if not ask Gissio - he is super helpful. (STLink adaptor & 50 pack of headers from Amazon for less than £10)

**Reply by gc01noob** [reply]
seems a bit weird that it won't let you overwrite - has it changed the filename on the GC01 drive after the first copy? it should just overwrite and prompt to confirm, then boot radpro - if it does maybe try renaming the file to match before the 2nd copy - keep the window open between copies - the filename doesn't matter it looks for the extension - maybe the original filename is too long and it truncates it
the .txt file is a status note and its name can change if there's a bootloader error

seems yours is slightly different but at least you're making progress, especially that your PC sees the USB device
if you look back to last month-ish i had a discussion with a guy who did the copy using the terminal only and he had to use a specific command to get it to overwrite - i can't remember it exactly but worth a try

**Reply by gc01noob** [reply]
btw are you sure you're using the correct firmware for your MCU?

**Reply by gc01noob** [reply]
look inside the full repo zip file, in documents>gc01 you'll find an original firmware - copy that twice and you should get it to boot again with the original system

**Comment by kd5byb** [comment]
Evening all!

The oddness continues.  The textfile at one point did go to "APPERR" at one point, but has stayed at "ReadMe.txt" many attempts after that.  For sure, I'm using the right file, I've got the CH32F103R8 (WCH) microprocessor.  What's REALLY ODD NOW is that the STLINK dongle doesn't even find the processor.  The screen also stopped working.  I think I've had totally messed up this unit.  I've had it for at least a year now and had a good bit of fun with it, so can't say I'm too upset if it is indeed dead now.  All of this is very, very strange!

**Reply by gc01noob** [reply]
forgot how important this is: make sure your internal backup battery is a healthy voltage or the firmware updates etc will fail - it's worth replacing the original for a new one. can get on ebay for a few £/$ (CR1220 i think, says on the button cell)

**Reply by kd5byb** [reply]
Oh wow - the battery in mine was stone cold dead.  Unfortunately, after replacing it with a fresh new one, I'm having the same issues as above.  I can see the GC01 BOOT drive in Linux, can copy the new firmware over twice, and it won't reboot, and won't load.  Using the STLINK I get the same "halted due to debug request."  Screen stays blank the whole time, so I'm pretty sure I've somehow smoked my unit.  :(

**Reply by gc01noob** [reply]
are you able to flash one of the standard/older firmware versions from the repo doc folder?
did you try a different cable?
have you attempted to reset after no main battery connected for a while?
start from scratch and retry all your actions again
smetimes these can kick it back into life

gissio might be able to help, he checks in every couple of weeks.

**Reply by jonwhite227** [reply]
My GC-01 was apparently completely 'bricked', but I got it working again with help from Gissio. He is super helpful. 

Have you tried flashing with main battery disconnected?

**Reply by ChyrkaA** [reply]
> My GC-01 was apparently completely 'bricked', but I got it working again with help from Gissio. He is super helpful.
> 
> Have you tried flashing with main battery disconnected?

Why Gissio not write for me in my disscussion(?

**Reply by jonwhite227** [reply]
I'm sure he will help you if he can, but he is in a different time zone - US I think - give it time, there is life outside GitHub.😉

**Comment by ChyrkaA** [comment]
> The long-awaited Rad Pro 2.0 is out, with support for several devices: FS2011, Bosean FS-600, FS-1000, FS-5000 and FNIRSI GC-01.
> 
> The UI has been updated to the [OpenBridge 5.0](https://www.openbridge.no/) design system, improving legibility under sunlight.
> 
> The default conversion factor for J613/J614 tubes has also been changed to 30 cpm/µSv/h, improving accuracy.
> 
> The FNIRSI GC-01 display panel settings (added in 2.0rc5) have been removed. And the granularity and range of the "SET tubeHVFrequency" and "SET tubeHVDutyCycle" data communication requests have been increased, to CPU clock resolution and 100 Hz-100 kHz and 0 %-100 %, respectively (not stored on power-off).
> 
> Enjoy!

Hello. Could you help me with my problem that I described in the next discussion? I kindly ask you to give me some time.

**Reply by Gissio** [reply]
Replied to in the other thread.
